src/Microsoft.ML.Core/ComponentModel/ComponentCatalog.cs

```diff
@@ -185,16 +185,25 @@ internal LoadableClassInfo(LoadableClassAttributeBase attr, MethodInfo getter, C
             internal object CreateInstanceCore(object[] ctorArgs)
             {
                 Contracts.Assert(Utils.Size(ctorArgs) == CtorTypes.Length + ((RequireEnvironment) ? 1 : 0));
-
-                if (InstanceGetter != null)
+                try
                 {
-                    Contracts.Assert(Utils.Size(ctorArgs) == 0);
-                    return InstanceGetter.Invoke(null, null);
+                    if (InstanceGetter != null)
+                    {
+                        Contracts.Assert(Utils.Size(ctorArgs) == 0);
+                        return InstanceGetter.Invoke(null, null);
+                    }
+                    if (Constructor != null)
+                        return Constructor.Invoke(ctorArgs);
+                    if (CreateMethod != null)
+                        return CreateMethod.Invoke(null, ctorArgs);
+                }
+                catch (Exception ex)
+                {
+                    if (ex.InnerException != null && ex.InnerException.IsMarked())
+                        throw Contracts.Except(ex, "Error during class instantiation");
```

Comment:
Does this actually add value? Why not just let the original exception be thrown? It will have the stacktrace of where it came from.

Reply:
Because of `Invoke` call. If anything happens inside it, we got TargetInvocationException and original exception would be inside InnerException.

Reply:
At least `catch (TargetInvocationException ex)` here then.